exec/java-exec/src/test/java/org/apache/drill/exec/store/bson/TestBsonRecordReader.java

```diff
@@ -45,21 +46,24 @@
 import org.bson.BsonTimestamp;
 import org.bson.BsonWriter;
 import org.bson.types.ObjectId;
-import org.junit.AfterClass;
-import org.junit.BeforeClass;
+import org.junit.After;
+import org.junit.Before;
 import org.junit.Test;
 
-public class TestBsonRecordReader extends BaseTestQuery {
-  private static VectorContainerWriter writer;
-  private static TestOutputMutator mutator;
-  private static BsonRecordReader bsonReader;
+public class TestBsonRecordReader {
+  private BufferAllocator allocator;
+  private VectorContainerWriter writer;
+  private TestOutputMutator mutator;
+  private DrillBuf buffer;
+  private BsonRecordReader bsonReader;
 
-  @BeforeClass
-  public static void setUp() {
-    BufferAllocator bufferAllocator = getDrillbitContext().getAllocator();
-    mutator = new TestOutputMutator(bufferAllocator);
+  @Before
+  public void setUp() {
+    allocator = new RootAllocator(400_000);
```

Comment:
Two minor suggestions:
- use `RootAllocator` with an unlimited amount of memory as limiting it to 400K does not test `BsonRecordReader` and depends more on how Vectors are allocated.
- instead of working with `DrillBuf` directly, use `BufferManager`.

Reply:
@vrozov Made the changes